src/SqsConsumer.ts

```diff
@@ -0,0 +1,181 @@
+      const status = await this.processMessage(eventBridgeContent);
+
+      if (!status) {
+        console.log(`adding to DLQ -> ${JSON.stringify(data.Messages[0])}`);
+        await this.insertToDLQ(data.Messages[0]);
+      }
+
+      //delete all messages as they are moved to DLQ
+      await this.deleteMessage(data.Messages[0]);
+
+      // Schedule next message poll
+      await this.scheduleNextPoll(
+        config.aws.sqs.sharedSnowplowQueue.afterMessagePollIntervalSeconds *
+          1000
+      );
+    } else {
+      // If no messages were found, schedule another poll after 5 mins
+      await this.scheduleNextPoll(
+        config.aws.sqs.sharedSnowplowQueue.defaultPollIntervalSeconds * 1000
+      );
+    }
+  }
+
+  /**
+   * returns false if it cannot process the given message
+   * any error is logged in sentry
+   * @param event, event bridge content from SNS payload
+   * @return true if processed successfully, false if error occurs
+   */
+  async processMessage(event: any) {
+    try {
+      const detailType = event['detail-type'];
+
+      if (eventConsumer[detailType] == null) {
+        throw new Error(
+          `Unable to retrieve handler for detailType='${detailType}'`
+        );
+      }
+
+      await eventConsumer[detailType](event);
+      return true;
+    } catch (error) {
+      const errorMessage = 'Error deleting message from queue';
+      console.error(errorMessage, error);
+      console.error(JSON.stringify(event));
+      Sentry.addBreadcrumb({ message: errorMessage, data: event });
+      Sentry.captureException(error);
+      return false;
+    }
+  }
+
+  /**
+   * schedules the next polling
+   * @param timeout
+   */
+  async scheduleNextPoll(timeout: number) {
+    if (timeout > 0) {
+      await setTimeout(timeout);
+    }
+    this.emitter.emit(SqsConsumer.eventName);
+  }
+
+  /**
+   * Delete a message that has been handled from the
+   * snowplow event queue
+   * @param message processed SQS message
+   */
+  private async deleteMessage(message: Message) {
+    console.log(`deleting SQS message -> ` + JSON.stringify(message));
+
+    const deleteParams = {
+      QueueUrl: config.aws.sqs.sharedSnowplowQueue.url,
+      ReceiptHandle: message.ReceiptHandle,
+    };
+    try {
+      await this.sqsClient.send(new DeleteMessageCommand(deleteParams));
+    } catch (error) {
+      const errorMessage = 'Error deleting message from queue';
+      console.error(errorMessage, error);
+      console.error(JSON.stringify(message));
+      Sentry.addBreadcrumb({ message: errorMessage, data: message });
+      Sentry.captureException(error);
+    }
+  }
+
+  private async insertToDLQ(message) {
+    const insertParams = {
+      QueueUrl: config.aws.sqs.sharedSnowplowQueue.dlqUrl,
+      MessageBody: message.Body,
+    };
+    try {
+      await this.sqsClient.send(new SendMessageCommand(insertParams));
+    } catch (error) {
+      const errorMessage = 'Error deleting message from queue';
```

Comment:
```suggestion
      const errorMessage = 'Error inserting message from queue';
```

ditto to above. guessing this is copypasta